lib/utils/request.ts

```diff
@@ -1,18 +1,50 @@
-export type RequestFnType = (url: string, data: object) => void;
+export type RequestCallback = (error?: any, data?: any) => void;
+export type RequestFnType = (
+  url: string,
+  data: object,
+  callback?: RequestCallback
+) => void;
 
-export const requestWithSendBeacon: RequestFnType = (url, data) => {
+export const requestWithSendBeacon: RequestFnType = (url, data, callback) => {
   const serializedData = JSON.stringify(data);
   navigator.sendBeacon(url, serializedData);
+  if (callback) {
+    callback();
```

Comment:
Here the callback is deceiving I think, because it's not actually being called after the requests, and so you cannot document it as 

> this function gets called when the request is done, and exposes the status


I think it might be a better DX to show a warning explaining we're in the browser and using sendBeacon guaranteeing calls are always fired but not being able to check it with a callback, and not call the callback at all.

Reply:
Yeah it's indeed deceiving because we don't get any callback from sendBeacon. However, I think we should still call the callback for sendBeacon too, because what if they do some logic after sending the events? In the browsers that don't support sendBeacon, it will fallback to XMLHttpRequest, so we'd better have a consistent behavior. What do you think?

Reply:
we're essentially simulating the callback. It'd be ok if the simulation worked like an actual callback, firing after the requests, however it's not the case here. So the behaviour is radically different based on whether sendBeacon or XMLHttpRequest was picked, internally, and it's essentially a race condition hard to detect. This is what makes me uncomfortable with the API. 

I think as a user, I would prefer not having the callback fire at all and have a warning explaining to be why.